src/status_im/chat/models/images.cljs

```diff
@@ -175,13 +177,20 @@
 (rf/defn image-selected
   {:events [:chat.ui/image-selected]}
   [{:keys [db]} current-chat-id original uri]
-  {:db (update-in db [:chat/inputs current-chat-id :metadata :sending-image original] merge {:uri uri})})
+  {:db (-> db
+           (update-in [:chat/inputs current-chat-id :metadata :sending-image original] merge {:uri uri})
+           (update-in [:chat/inputs current-chat-id :metadata :selected-photos] conj original))})
 
 (rf/defn image-unselected
   {:events [:chat.ui/image-unselected]}
   [{:keys [db]} original]
-  (let [current-chat-id (:current-chat-id db)]
-    {:db (update-in db [:chat/inputs current-chat-id :metadata :sending-image] dissoc original)}))
+  (let [current-chat-id                (:current-chat-id db)
+        selected-photos-after-deletion (->> (rf/sub [:chats/selected-photos])
```

Comment:
@ibrkhalil, We should never use subscriptions inside event handlers. Actually the rule is simpler, only use subscriptions in UI components. Whenever a subscription call is made, it causes overhead in re-frame, because it thinks it needs to re-run the function that's deref'ing the subscription, but re-running the event handler makes no sense.